src/sentry/models/repository.py

```diff
@@ -2,22 +2,52 @@
 
 from django.db import models
 from django.utils import timezone
+from jsonfield import JSONField
 
+from sentry.constants import ObjectStatus
 from sentry.db.models import (
     BoundedPositiveIntegerField, Model, sane_repr
 )
+from sentry.signals import pending_delete
 
 
 class Repository(Model):
     __core__ = True
 
     organization_id = BoundedPositiveIntegerField(db_index=True)
     name = models.CharField(max_length=200)
+    url = models.URLField(null=True)
+    provider = models.CharField(max_length=64, null=True)
+    external_id = models.CharField(max_length=64, null=True)
+    config = JSONField(default=lambda: {})
+    status = BoundedPositiveIntegerField(
+        default=ObjectStatus.VISIBLE,
+        choices=ObjectStatus.as_choices(),
+        db_index=True,
```

Comment:
_shouldn't_ need an index here since you're always querying for `WHERE status=VISIBLE` or `WHERE status != VISIBLE`, which shouldn't use an index since 99% of rows will match this. The planner is going to choose to filter the rows anyways.